migrating_from_ocp_3_to_4/migrating-applications-3-4.adoc

```diff
@@ -1,4 +1,4 @@
-[id="migrating-applications-3-4_{context}"]
+[id="migrating-applications-3-4"]
```

Comment:
I also don't think you need line 5 below for anything:

> :migrating-applications-3-4:

That should already be covered by `context` in line 4.

Reply:
Actually, you do. The 'ifdefs' in modules do not work unless you add the `:migrating-applications-3-4:` after `:context: migrating-applications-3-4`. Try it and you will see what I mean.

Reply:
Ah, my bad. Thanks for the explanation!

Reply:
It's one of those "gotchas" that no one ever seems to mention. In most cases, ":context: migration" works just fine -- the links all show the context correctly. But if you try to use ifdefs in any of the modules, the ifdef sections do not show up unless you also use ":migration:" AND ":migration!:" to enclose the included modules. Also very important to remember to turn off the explicit context because otherwise the context can be propagated to the next assembly.